frontend/src/metabase-types/api/mocks/dataset.ts

```diff
@@ -21,24 +22,29 @@ export type MockDatasetOpts = Partial<Omit<Dataset, "data">> & {
 export const createMockDataset = ({
   data = {},
   ...opts
-}: MockDatasetOpts = {}) => ({
-  data: {
-    rows: [],
-    cols: [
-      createMockColumn({
-        display_name: "NAME",
-        source: "native",
-        name: "NAME",
-      }),
-    ],
-    rows_truncated: 0,
-    ...data,
-  },
-  database_id: 1,
-  row_count: 0,
-  running_time: 1000,
-  ...opts,
-});
+}: MockDatasetOpts = {}) => {
+  const columns = data.cols ?? [
```

Comment:
@ranquild Let me know if this is what you had in mind - I figured that we want to use the same column data for the results metadata and `data.cols` since that's what I'm seeing in the model's API return value